sklearn/ensemble/_weight_boosting.py

```diff
@@ -1089,7 +1201,7 @@ def _boost(self, iboost, X, y, sample_weight, random_state):
         # for all samples in the training set
         X_ = _safe_indexing(X, bootstrap_idx)
         y_ = _safe_indexing(y, bootstrap_idx)
-        estimator.fit(X_, y_)
+        estimator.fit(X_, y_, **fit_params)
```

Comment:
The current behavior is that the regressor never passes sample_weights to the base_estimator.